waspc/packages/deploy/src/providers/fly/DeploymentInfo.ts

```diff
@@ -0,0 +1,49 @@
+import { ITomlFilePaths } from './helpers/tomlFileHelpers.js'
+import { IGlobalOptions } from './IGlobalOptions.js'
+
+export interface IDeploymentInfo {
```

Comment:
Ah, interesting, thanks for the idea. I will take a look and probably swap it out with what you have (since it is mostly just readonly data). I wasn't sure how heavily I should rely on one or the other in TS, so seeing the options is helpful. 👍🏻

Reply:
Great points by Miho.

Here are a few general rules of thumb for using vs not using classes in TypeScript:
- If all methods are static, it shouldn't be a class
- If there's no internal state, it should't be a class
- If there are non-static (instance) methods and there is some internal state, it could either be a class or a closure (sometimes one is more appropriate than the other)

Classes are mainly useful:
- When dealing with complex hierarchies
- When we want to use popular OO design patterns
- When we need a blueprint for objects with complex internal state

So, in short, we almost never need them :)

Reply:
Yeah functional crowd go go :D